test/Dialect/SMT/basic.mlir

```diff
@@ -5,3 +5,51 @@
 func.func @types(%arg0: !smt.bool, %arg1: !smt.bv<32>) {
   return
 }
+
+func.func @core(%in: i8) {
+  // CHECK: %a = smt.declare_const "a" {smt.some_attr} : !smt.bool
+  %a = smt.declare_const "a" {smt.some_attr} : !smt.bool
+  // CHECK: smt.declare_const {smt.some_attr} : !smt.bv<32>
+  %b = smt.declare_const {smt.some_attr} : !smt.bv<32>
+
+  // CHECK: smt.assert %a {smt.some_attr}
+  smt.assert %a {smt.some_attr}
```

Comment:
Just to be clear, this supposedly has undefined semantics, right?
As there is no toplevel `solver` here.

Reply:
It depends. In this file, no entry point function is defined, so if you insert a `main` function that has an `smt.solver` that has a call to this function in its body, it would be well-defined. If you run this function directly, it's undefined (in practice, it will crash because no SMT context is initialized). I didn't focus on this here because it's just a regression test for the parser and printer, but I can add one if you want.

I was also thinking about requiring all smt operations being (transitive) children of the `smt.solver` operation, but that would make things a lot more rigid (making it more difficult to reuse code, forcing inlining of everything, etc.). On the positive side, we could properly verify a context is there.